packages/core/src/delegation.js

```diff
@@ -162,16 +162,46 @@ export class Delegation {
     this.root = root
     this.blocks = blocks
 
-    /** @type {API.AttachedLinkSet} */
-    this.attachedLinks = new Set()
-
     Object.defineProperties(this, {
       blocks: {
         enumerable: false,
       },
     })
   }
 
+  /**
+   * @returns {API.AttachedLinkSet}
+   */
+  get attachedLinks() {
+    const _attachedLinks = new Set()
+    const ucanView = decode(this.root)
+
+    // Get links from capabilities nb
+    for (const capability of ucanView.capabilities) {
+      /** @type {Link[]} */
+      const links = Object.values(capability.nb || {})
```

Comment:
I think it might be a good idea to do a full traversal to capture nested links e.g. because we have couple of instances where that is the case